src/hera/workflow.py

```diff
@@ -193,8 +189,9 @@ def __init__(
             setattr(self.spec, "image_pull_secrets", secret_refs)
 
         if self.affinity:
-            setattr(self.exit_template, "affinity", self.affinity.get_spec())
```

Comment:
nit: follow the same order when setting attributes (spec -> dag -> template -> exit_template).